CMakeLists.txt

```diff
@@ -235,61 +317,31 @@ add_library(lego1 SHARED
   LEGO1/omni/src/video/mxvideoparam.cpp
   LEGO1/omni/src/video/mxvideoparamflags.cpp
   LEGO1/omni/src/video/mxvideopresenter.cpp
-  LEGO1/realtime/matrix.cpp
-  LEGO1/realtime/orientableroi.cpp
-  LEGO1/realtime/realtime.cpp
   LEGO1/realtime/realtimeview.cpp
```

Comment:
`link.exe` does only export symbols from objects, not from static libraries.
Because realtimeview.cpp exports some symbols, it cannot be added to a realtime static library.
So perhaps this source belongs in LegoOmni instead?

Reply:
Maybe I misunderstand, but doesn't it seem to work? Otherwise there should be an error in the verexp tool regarding a missing symbol?

Reply:
In 76debb7e5b9c9ce177acde36849c415291ce27d6, I moved it back from realtime to lego01.

Reply:
ok, let me see what the best course of action is here. The 1996 has it as its own library, but maybe they changed it

Reply:
Hmm, if there really is no other way to make this work we'll have to leave it like that I guess. The odd thing is that the functions in `realtimeview.cpp` appear in between several other static libs, so if this was really part of LegoOmni then everything that comes after it would also have to have been compiled as part of LegoOmni, which seems unlikely.

Reply:
I'm assuming using a .DEF file wouldn't make a difference?
https://learn.microsoft.com/en-us/cpp/build/reference/module-definition-dot-def-files?view=msvc-170

Reply:
A `.def` could work, but that will only work for MSVC.
mingw mangles the c++ symbols differently.

Reply:
It it possible to build omni and legoomni as static libraries, but at the end, only their objects will be used (not the .lib)

Reply:
Based on the 1996 sources I think only Omni was built as a library, and then linked at the very end.

Roughly speaking, we are seeing the code in this order:

1) LegoOmni
2) MSVC libraries
3) Top-level libraries and lego/sources libraries
4) Omni

maybe @tahg could give us a default accounting of especially the libraries in the middle since he has a spreadsheet with everything

Reply:
Makes sense, if and only if the omni libraries don't export. Otherwise they would had to use some kind of `.def`.

Reply:
Considering I can't find a single `dllexport` in the 1996 sources I have a feeling they probably did use those .def files instead. Omni does export a lot of functions, so I can't think of any other way

Reply:
It makes also sense because the number of exported symbols is kept to the very minimum.
They were very careful to not export too many symbols.

Reply:
We don't have source for anything that was in a DLL as far as I know.  If you look at the LEGOAPP folder, you can see that quite a lot more was in the main exe.

Reply:
Because mingw's thiscall and msvc's thiscall convention are incompatible, msvc and mingw have their own `.def` file.
I removed the `COMPAT_CONST` version of `MxVideoParam::MxVideoParam` so current MSVC can keep using the same def as msvc4.

Can you reformat?

Reply:
Ran the format.

I've found a few more "complete" link instructions from the 1996 sources:

`LEGO/LEGOOMNI/SRC/DLL/LEGOOMNI.MAK:# ADD LINK32 omni2dSH.lib libcmt.lib winmm.lib dsound.lib kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib miscd.lib roid.lib animd.lib geomd.lib tglrl40d.lib d3drm.lib 3dmanagerd.lib realtimed.lib viewmanagerd.lib pcubed.lib oldnames.lib /nologo /subsystem:windows /dll /debug /machine:I386 /nodefaultlib /out:"..\..\bin\LegoOmni1dSH.dll" /implib:"..\..\lib\LegoOmni1dSH.lib"`

and

`LEGO/LEGOOMNI/SRC/DLL/LEGOOMNI.MAK:# ADD BASE LINK32 libcmt.lib winmm.lib omni2d.lib dsound.lib kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib miscd.lib roid.lib animd.lib geomd.lib tglrl20d.lib rlddf.lib rlddfglu.lib rlff.lib 3dmanagerd.lib realtimed.lib viewmanagerd.lib pcubed.lib /nologo /subsystem:windows /dll /debug /machine:I386 /nodefaultlib /out:"..\..\bin\LegoOmni1d.dll" /implib:"..\..\lib\LegoOmni1d.lib"`

It would seem that they generate a separate file with `SH` appended for the SmartHeap version, which is a detail we could cover but not too important.

We stilll need to add yet another library called `3dmanager` based on `lego/sources/3dmanager`. At that point we should have all of it (for now).

Moving `realtimeview.cpp` to the the realtime lib seems to work now with the `.DEF` files.

Reply:
Also found the reference to the `.DEF` they used:

`LEGO/LEGOOMNI/SRC/DLL/LEGOOMNI.DEF:; LegoOmni.def : Declares the module parameters for the OMNI.DLL.`

Reply:
Let's ignore the `SH` for now, because that would also mean those libraries are built without smartheap's `operator new`.
For example, does that mean 3dmanager is always built without smartheap?
Mixing allocators is tricky.

Reply:
The various `__declspec(dllexport)`'s can also be removed now.

Reply:
I've removed them, but this caused the msys build to break it seems. Otherwise this looks pretty good now.